Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -0,0 +1,316 @@
+        case off = "option-off"
+
+        var settingTitle: String {
+            switch self {
+            case .on:
+                return Strings.TrackingProtectionOptionAlwaysOn
+            case .onInPrivateBrowsing:
+                return Strings.TrackingProtectionOptionOnInPrivateBrowsing
+            case .off:
+                return Strings.TrackingProtectionOptionAlwaysOff
+            }
+        }
+
+        static let allOptions: [EnabledOption] = [.on, .onInPrivateBrowsing, .off]
+    }
+
+    // Raw values are stored to prefs, be careful changing them.
+    enum StrengthOption: String {
+        case basic = "option-basic"
+        case strict = "option-strict"
+
+        var settingTitle: String {
+            switch self {
+            case .basic:
+                return Strings.TrackingProtectionOptionBlockListTypeBasic
+            case .strict:
+                return Strings.TrackingProtectionOptionBlockListTypeStrict
+            }
+        }
+
+        var subtitle: String {
+            switch self {
+            case .basic:
+                return Strings.TrackingProtectionOptionBlockListTypeBasicDescription
+            case .strict:
+                return Strings.TrackingProtectionOptionBlockListTypeStrictDescription
+            }
+        }
+
+        static let allOptions: [StrengthOption] = [.basic, .strict]
+    }
+
+    static func prefsChanged() {
+        NotificationCenter.default.post(name: Notification.Name(rawValue: NotificationContentBlockerReloadNeeded), object: nil)
+    }
+
+    class func name() -> String {
+        return "ContentBlockerHelper"
+    }
+
+    private static var heavyInitHasRunOnce = false
+
+    init(tab: Tab, profile: Profile) {
+        self.ruleStore = WKContentRuleListStore.default()
+        if self.ruleStore == nil {
+            print("WKContentRuleListStore unavailable.")
+            assert(false)
+            return
+        }
+
+        self.tab = tab
+        self.profile = profile
+
+        NotificationCenter.default.addObserver(self, selector: #selector(ContentBlockerHelper.reloadTab), name: NSNotification.Name(rawValue: NotificationContentBlockerReloadNeeded), object: nil)
+
+        addActiveRulesToTab(reloadTab: false)
+
+        if ContentBlockerHelper.heavyInitHasRunOnce {
+            return
+        }
+        ContentBlockerHelper.heavyInitHasRunOnce = true
+
+        var wasRuleListChangedDuringInit = false
+        removeOldListsByDateFromStore() { isListChanged in
+            wasRuleListChangedDuringInit = isListChanged
+
+            self.removeOldListsByNameFromStore() { isListChanged in
+                wasRuleListChangedDuringInit = wasRuleListChangedDuringInit || isListChanged
+
+                self.compileListsNotInStore() { isListChanged in
+                    wasRuleListChangedDuringInit = wasRuleListChangedDuringInit || isListChanged
+
+                    if wasRuleListChangedDuringInit {
```

Comment:
have time to switch this over before next beta, will switch it over and i'll get more practice with deferredness